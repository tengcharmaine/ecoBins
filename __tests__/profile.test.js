// import React from 'react';
// import renderer from 'react-test-renderer';
// import HomeScreen from '../components/profile';
// import { render } from '@testing-library/react-native';

// describe('Profile', () => {
//   test('renders correctly', () => {
//     const tree = renderer.create(<HomeScreen />).toJSON();
    
//     // Check if the required elements are rendered
//     expect(tree).toMatchSnapshot();
//   });

//   test('displays the username', () => {
//     const username = 'John Doe';
//     const { getByText } = render(<HomeScreen username={username} />);
//     const usernameText = getByText(/Welcome, John Doe!/i);
//     expect(usernameText).toBeTruthy();
//   });

//   test('displays the remaining points', () => {
//     const remainingPoints = 100;
//     const { getByText } = render(<HomeScreen remainingPoints={remainingPoints} />);
//     const pointsText = getByText(/You have 100 points accumulated so far./i);
//     expect(pointsText).toBeTruthy();
//   });
// });

import React from 'react';
import { render, fireEvent } from '@testing-library/react-native';
import renderer from 'react-test-renderer';
import HomeScreen from '../components/profile';

jest.mock('../lib/supabase', () => ({
    supabase: {
      auth: {
        getUser: jest.fn().mockResolvedValue({
          data: { user: { id: 'mockUserId' } },
        }),
      },
      from: () => ({
        select: jest.fn().mockReturnThis(),
        eq: jest.fn().mockReturnThis(),
        single: jest.fn().mockResolvedValue({
          data: { email: 'johndoe@example.com', profile: 'mockProfilePicture' },
        }),
      }),
    },
  }));

describe('Profile', () => {
  test('renders correctly', () => {
    const tree = renderer.create(<HomeScreen />).toJSON();
    expect(tree).toMatchSnapshot();
  });

//   test('displays the username in the welcome message', async () => {
//     const { getByText } = render(<HomeScreen />);
//     const welcomeText = getByText('Welcome, johndoe@example.com!');
//     expect(welcomeText).toBeTruthy();
//   });

//   test('displays the remaining points', () => {
//     const { getByText } = render(<HomeScreen />);
//     const remainingPoints = 100;
//     const pointsText = getByText(`You have ${remainingPoints} points accumulated so far.`);
//     expect(pointsText).toBeTruthy();
//   });

//   test('calls navigation.navigate when edit profile picture icon is pressed', () => {
//     const navigate = jest.fn();
//     const { getByTestId } = render(<HomeScreen navigation={{ navigate }} />);
//     const editProfilePictureIcon = getByTestId('edit-profile-picture-icon');
//     fireEvent.press(editProfilePictureIcon);
//     expect(navigate).toHaveBeenCalledWith('editprofilepic');
//   });

//   test('calls navigation.navigate when edit username icon is pressed', () => {
//     const navigate = jest.fn();
//     const { getByTestId } = render(<HomeScreen navigation={{ navigate }} />);
//     const editUsernameIcon = getByTestId('edit-username-icon');
//     fireEvent.press(editUsernameIcon);
//     expect(navigate).toHaveBeenCalledWith('editusername');
//   });
});
