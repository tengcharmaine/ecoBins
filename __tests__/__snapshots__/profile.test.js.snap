// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Profile renders correctly 1`] = `
<View
  style={
    {
      "alignItems": "center",
      "flex": 1,
      "justifyContent": "center",
      "padding": 50,
    }
  }
>
  <View>
    <Text
      style={
        [
          {
            "textAlign": "left",
          },
          {
            "color": "rgba(28, 27, 31, 1)",
            "fontFamily": "System",
            "fontWeight": "400",
            "letterSpacing": 0,
          },
          {
            "writingDirection": "ltr",
          },
          undefined,
        ]
      }
    >
      No profile picture found
    </Text>
    <View
      collapsable={false}
      style={
        {
          "backgroundColor": "white",
          "borderRadius": 20,
          "bottom": 0,
          "height": 41,
          "margin": 6,
          "position": "absolute",
          "right": 0,
          "shadowColor": "#000",
          "shadowOffset": {
            "height": 0,
            "width": 0,
          },
          "shadowOpacity": 0,
          "shadowRadius": 0,
          "width": 41,
        }
      }
      testID="icon-button-container-outer-layer"
    >
      <View
        collapsable={false}
        style={
          {
            "backgroundColor": "white",
            "borderColor": "rgba(121, 116, 126, 1)",
            "borderRadius": 20,
            "borderWidth": 0,
            "elevation": 0,
            "flex": 1,
            "overflow": "hidden",
            "shadowColor": "#000",
            "shadowOffset": {
              "height": 0,
              "width": 0,
            },
            "shadowOpacity": 0,
            "shadowRadius": 0,
          }
        }
        testID="icon-button-container"
      >
        <View
          accessibilityComponentType="button"
          accessibilityRole="button"
          accessibilityState={
            {
              "busy": undefined,
              "checked": undefined,
              "disabled": false,
              "expanded": undefined,
              "selected": undefined,
            }
          }
          accessibilityTraits="button"
          accessibilityValue={
            {
              "max": undefined,
              "min": undefined,
              "now": undefined,
              "text": undefined,
            }
          }
          accessible={true}
          centered={true}
          collapsable={false}
          color="black"
          focusable={true}
          hitSlop={
            {
              "bottom": 6,
              "left": 6,
              "right": 6,
              "top": 6,
            }
          }
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            [
              {
                "overflow": "hidden",
              },
              [
                {
                  "alignItems": "center",
                  "flexGrow": 1,
                  "justifyContent": "center",
                },
                {
                  "borderRadius": 20,
                },
              ],
            ]
          }
          testID="icon-button"
        >
          <Text
            accessibilityElementsHidden={true}
            importantForAccessibility="no-hide-descendants"
            pointerEvents="none"
            selectable={false}
            style={
              [
                {
                  "backgroundColor": "transparent",
                },
                {
                  "color": "rgba(73, 69, 79, 1)",
                  "fontSize": 25,
                },
              ]
            }
          >
            □
          </Text>
        </View>
      </View>
    </View>
  </View>
  <View
    style={
      {
        "alignItems": "center",
        "flexDirection": "row",
        "marginVertical": 20,
      }
    }
  >
    <Text
      style={
        [
          {
            "textAlign": "left",
          },
          {
            "color": "rgba(28, 27, 31, 1)",
            "fontFamily": "System",
            "fontWeight": "400",
            "letterSpacing": 0,
          },
          {
            "writingDirection": "ltr",
          },
          {
            "fontSize": 16,
          },
        ]
      }
    >
      Welcome, 
      !
    </Text>
    <View
      collapsable={false}
      style={
        {
          "backgroundColor": "transparent",
          "borderRadius": 18,
          "height": 36,
          "margin": 6,
          "shadowColor": "#000",
          "shadowOffset": {
            "height": 0,
            "width": 0,
          },
          "shadowOpacity": 0,
          "shadowRadius": 0,
          "width": 36,
        }
      }
      testID="icon-button-container-outer-layer"
    >
      <View
        collapsable={false}
        style={
          {
            "backgroundColor": "transparent",
            "borderColor": "rgba(121, 116, 126, 1)",
            "borderRadius": 18,
            "borderWidth": 0,
            "elevation": 0,
            "flex": 1,
            "overflow": "hidden",
            "shadowColor": "#000",
            "shadowOffset": {
              "height": 0,
              "width": 0,
            },
            "shadowOpacity": 0,
            "shadowRadius": 0,
          }
        }
        testID="icon-button-container"
      >
        <View
          accessibilityComponentType="button"
          accessibilityRole="button"
          accessibilityState={
            {
              "busy": undefined,
              "checked": undefined,
              "disabled": false,
              "expanded": undefined,
              "selected": undefined,
            }
          }
          accessibilityTraits="button"
          accessibilityValue={
            {
              "max": undefined,
              "min": undefined,
              "now": undefined,
              "text": undefined,
            }
          }
          accessible={true}
          centered={true}
          collapsable={false}
          color="black"
          focusable={true}
          hitSlop={
            {
              "bottom": 6,
              "left": 6,
              "right": 6,
              "top": 6,
            }
          }
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            [
              {
                "overflow": "hidden",
              },
              [
                {
                  "alignItems": "center",
                  "flexGrow": 1,
                  "justifyContent": "center",
                },
                {
                  "borderRadius": 18,
                },
              ],
            ]
          }
          testID="icon-button"
        >
          <Text
            accessibilityElementsHidden={true}
            importantForAccessibility="no-hide-descendants"
            pointerEvents="none"
            selectable={false}
            style={
              [
                {
                  "backgroundColor": "transparent",
                },
                {
                  "color": "rgba(73, 69, 79, 1)",
                  "fontSize": 20,
                },
              ]
            }
          >
            □
          </Text>
        </View>
      </View>
    </View>
  </View>
  <Text
    style={
      [
        {
          "textAlign": "left",
        },
        {
          "color": "rgba(28, 27, 31, 1)",
          "fontFamily": "System",
          "fontWeight": "400",
          "letterSpacing": 0,
        },
        {
          "writingDirection": "ltr",
        },
        {
          "fontSize": 16,
        },
      ]
    }
  >
    You have 
    0
     points accumulated so far.
  </Text>
  <View
    collapsable={false}
    style={
      {
        "backgroundColor": "#c7dede",
        "borderRadius": 10,
        "marginBottom": 10,
        "marginTop": 20,
        "shadowColor": "#000",
        "shadowOffset": {
          "height": 0,
          "width": 0,
        },
        "shadowOpacity": 0,
        "shadowRadius": 0,
        "width": "25%",
      }
    }
    testID="button-container-outer-layer"
  >
    <View
      collapsable={false}
      style={
        {
          "alignItems": "center",
          "backgroundColor": "#c7dede",
          "borderColor": "black",
          "borderRadius": 10,
          "borderStyle": "solid",
          "borderWidth": 0,
          "flex": undefined,
          "minWidth": 64,
          "shadowColor": "#000",
          "shadowOffset": {
            "height": 0,
            "width": 0,
          },
          "shadowOpacity": 0,
          "shadowRadius": 0,
        }
      }
      testID="button-container"
    >
      <View
        accessibilityRole="button"
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": true,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          [
            {
              "overflow": "hidden",
            },
            {
              "borderRadius": 10,
            },
          ]
        }
        testID="button"
      >
        <View
          style={
            [
              {
                "alignItems": "center",
                "flexDirection": "row",
                "justifyContent": "center",
              },
              undefined,
            ]
          }
        >
          <Text
            numberOfLines={1}
            selectable={false}
            style={
              [
                {
                  "fontFamily": "System",
                  "fontSize": 14,
                  "fontWeight": "500",
                  "letterSpacing": 0.1,
                  "lineHeight": 20,
                },
                {
                  "textAlign": "left",
                },
                {
                  "color": "rgba(28, 27, 31, 1)",
                  "writingDirection": "ltr",
                },
                [
                  {
                    "marginHorizontal": 16,
                    "marginVertical": 9,
                    "textAlign": "center",
                  },
                  false,
                  {
                    "marginHorizontal": 12,
                  },
                  undefined,
                  false,
                  {
                    "color": "rgba(103, 80, 164, 1)",
                    "fontFamily": "System",
                    "fontSize": 14,
                    "fontWeight": "500",
                    "letterSpacing": 0.1,
                    "lineHeight": 20,
                  },
                  undefined,
                ],
              ]
            }
            testID="button-text"
          >
            Logout
          </Text>
        </View>
      </View>
    </View>
  </View>
</View>
`;
